Refactor data persistence layer to improve quality and usability - Closes #2669 (#2673)
LiskHQ/lisk: storage/README.md

* Timestamp types always refer to the network Epoch timestamp. In the case of Unix timestamp, the attribute name should be prefixed with `unix_`.
* All values related to the amounts should be exposed as strings, due to the limited JavaScript integers precision (find out more while reading about Number. MAX_SAFE_INTEGER). The block height and round number can be safely stored as integers and should be exposed as integers.
* All `BYTEA` fields in 'hex' encoding should be exposed as strings. Any other encoding should be exposed as Buffer.

> * When required to use constants, use `Symbol` and then export it from interface, so later everyone explicitly use those defined constants instead of strings.

Reviewer: ```suggestion
* Use `Symbol` to define constants within database entities and export them, so the other modules can use them explicitly instead of strings.
```

> * When required to use constants, use `Symbol` and then export it from interface, so later everyone explicitly use those defined constants instead of strings.

Reviewer 2: ```suggestion
* Use `Symbol` to define constants within database entities and export them, so the other modules can use them explicitly instead of strings.
```
